backends/source/dppl_sycl_queue_interface.cpp

```diff
@@ -0,0 +1,457 @@
+ * @param    Device         My Param doc
+ */
+void dump_device_info (const device & Device)
+{
+    std::stringstream ss;
+
+    ss << std::setw(4) << " " << std::left << std::setw(16) << "Name"
+       << Device.get_info<info::device::name>() << '\n';
+    ss << std::setw(4) << " " << std::left << std::setw(16) << "Driver version"
+       << Device.get_info<info::device::driver_version>() << '\n';
+    ss << std::setw(4) << " " << std::left << std::setw(16) << "Vendor"
+       << Device.get_info<info::device::vendor>() << '\n';
+    ss << std::setw(4) << " " << std::left << std::setw(16) << "Profile"
+       << Device.get_info<info::device::profile>() << '\n';
+
+    std::cout << ss.str();
+}
+
+/*!
+ * @brief
+ *
+ * @param    Platform       My Param doc
+ */
+void dump_platform_info (const platform & Platform)
+{
+    std::stringstream ss;
+
+    ss << std::setw(4) << " " << std::left << std::setw(12) << "Name"
+       << Platform.get_info<info::platform::name>() << '\n';
+    ss << std::setw(4) << " " << std::left << std::setw(12) << "Version"
+       << Platform.get_info<info::platform::version>() << '\n';
+    ss << std::setw(4) << " " << std::left << std::setw(12) << "Vendor"
+       << Platform.get_info<info::platform::vendor>() << '\n';
+    ss << std::setw(4) << " " << std::left << std::setw(12) << "Profile"
+       << Platform.get_info<info::platform::profile>() << '\n';
+
+    std::cout << ss.str();
+}
+
+void error_reporter (const std::string & msg)
+{
+    throw std::runtime_error("Error: " + msg);
+}
+
+/*!
+ * @brief A helper class to support the DPPLSyclQueuemanager.
+ *
+ * The QMgrHelper is needed so that sycl headers are not exposed at the
+ * top-level DPPL API.
+ *
+ */
+class QMgrHelper
+{
+public:
+    static std::vector<cl::sycl::queue>   cpu_queues;
+    static std::vector<cl::sycl::queue>   gpu_queues;
+    static thread_local std::vector<cl::sycl::queue> active_queues;
+
+    static __dppl_give DPPLSyclQueueRef
+    getQueue (DPPLSyclDeviceType DeviceTy, size_t DNum);
+
+    static __dppl_give DPPLSyclQueueRef
+    getCurrentQueue ();
+
+    static void
+    setAsDefaultQueue (DPPLSyclDeviceType DeviceTy, size_t DNum);
+
+    static __dppl_give DPPLSyclQueueRef
+    setAsCurrentQueue (DPPLSyclDeviceType DeviceTy, size_t DNum);
+
+    static void
+    removeCurrentQueue ();
+
+    static cl::sycl::vector_class<cl::sycl::queue>
+    init_queues (info::device_type device_ty)
+    {
+        std::vector<cl::sycl::queue> queues;
+        for(auto d : device::get_devices(device_ty))
+            queues.emplace_back(d);
+        return queues;
+    }
+};
+
+// Initialize the active_queue with the default queue
+thread_local std::vector<cl::sycl::queue> QMgrHelper::active_queues
+    = {default_selector()};
+
+std::vector<cl::sycl::queue> QMgrHelper::cpu_queues
+    = QMgrHelper::init_queues(info::device_type::cpu);
+
+std::vector<cl::sycl::queue> QMgrHelper::gpu_queues
+    = QMgrHelper::init_queues(info::device_type::gpu);
```

Comment:
Make them static?

Reply:
These are already static members of `QMgrHelper`. I am initializing them here.